api/graphite.go

```diff
@@ -1048,11 +1087,19 @@ func (s *Server) clusterFindByTag(ctx context.Context, orgId uint32, expressions
 		}
 
 		// 0 disables the check, so only check if maxSeriesPerReq > 0
-		if maxSeriesPerReq > 0 && len(resp.Metrics)+len(allSeries) > maxSeries {
-			return nil,
-				response.NewError(
-					http.StatusRequestEntityTooLarge,
-					fmt.Sprintf("Request exceeds max-series-per-req limit (%d). Reduce the number of targets or ask your admin to increase the limit.", maxSeriesPerReq))
+		if maxSeries > 0 && len(resp.Metrics)+len(allSeries) > maxSeries {
+			if softLimit {
+				remainingSpace := maxSeries - len(allSeries)
+				// Fill in up to maxSeries
+				for _, series := range resp.Metrics[:remainingSpace] {
+					allSeries = append(allSeries, Series{
+						Pattern: series.Path,
+						Node:    r.peer,
+						Series:  []idx.Node{series},
+					})
+				}
+			}
+			return allSeries, TooManySeriesErr
```

Comment:
you want to excute this return even in the softlimit case?

Reply:
Yes. When soft limited, it just returns the results up to the limit

Reply:
but it will also error

Reply:
In the calling function we currently compare the error and ignore it. But I agree that it's cleaner to just not error.